pkg/cases/handler.go

```diff
@@ -8,14 +8,159 @@ import (
+}
+
+// @Summary Update case
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseDetail body string true "case ID and updated details"
+// @Success 200 {object} api.ReturnedData{}
+// @Router /patient/{patientID}/case/{caseID} [PUT]
+func (h *CaseHandler) UpdateCase(c echo.Context) error {
+	// ...
+	c.Logger().Debug("UpdateCase")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary New a prescrition
+// @Description
+// @Tags Case
+// @Produce json
+// @Param prescriptionDetail body string true "case ID and prescription details"
+// @Success 200 {object} api.ReturnedData{}
+// @Router /patient/{patientID}/case/{caseID}/prescription [POST]
+func (h *CaseHandler) NewPrescription(c echo.Context) error {
+	// ...
+	c.Logger().Debug("NewPrescription")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Delete a prescrition
+// @Description
+// @Tags Case
+// @Produce json
+// @Param prescriptionID path uint true "prescription ID"
+// @Success 200 {object} api.ReturnedData{}
+// @Router /patient/{patientID}/case/{caseID}/prescription/{prescriptionID} [DELETE]
+func (h *CaseHandler) DeletePrescription(c echo.Context) error {
+	// ...
+	c.Logger().Debug("DeletePrescription")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Update a prescrition
+// @Description
+// @Tags Case
+// @Produce json
+// @Param prescriptionDetails body uint true "prescription ID and updated details"
+// @Success 200 {object} api.ReturnedData{}
+// @Router /patient/{patientID}/case/{caseID}/prescription/{prescriptionID} [PUT]
+func (h *CaseHandler) UpdatePrescription(c echo.Context) error {
+	// ...
+	c.Logger().Debug("UpdatePrescription")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Get prescription by prescription id
+// @Description
+// @Tags Case
+// @Produce json
+// @Param prescriptionID path uint true "prescription ID"
+// @Success 200 {object} api.ReturnedData{data=Prescription}
+// @Router /patient/{patientID}/case/{caseID}/prescription/{prescriptionID}  [GET]
+func (h *CaseHandler) GetPrescriptionByPrescriptionID(c echo.Context) error {
+	// ...
+	c.Logger().Debug("GetPrescriptionByPrescriptionID")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Get prescriptions by case id
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseID path uint true "case ID"
+// @Success 200 {object} api.ReturnedData{data=[]Prescription}
+// @Router /patient/{patientID}/case/{caseID}/prescription  [GET]
+func (h *CaseHandler) GetPrescriptionByCaseID(c echo.Context) error {
+	// ...
+	c.Logger().Debug("GetPrescriptionByCaseID")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Query medicine by key word
+// @Description
+// @Tags Case
+// @Produce json
+// @Param keyword query string true "key word of the medicine"
+// @Success 200 {object} api.ReturnedData{data=[]Medicine}
+// @Router /medicine
+func (h *CaseHandler) QueryMedicine(c echo.Context) error {
```

Comment:
`GetMedicine` would be ok since you already put it in query. And `@param q query` is sufficient. Refer to [](https://github.com/typicode/json-server)